exporter/chronicleexporter/factory.go

```diff
@@ -0,0 +1,83 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chronicleexporter
+
+import (
+	"context"
+	"errors"
+
+	"github.com/observiq/bindplane-agent/exporter/chronicleexporter/internal/metadata"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+// TypeStr for the exporter type.
+const TypeStr = "chronicle"
+
+// Factory is the factory for the Chronicle exporter.
+type Factory struct {
+}
+
+// NewFactory creates a new Chronicle exporter factory.
+func NewFactory() exporter.Factory {
+	return exporter.NewFactory(
+		metadata.Type,
+		createDefaultConfig,
+		exporter.WithLogs(createLogsExporter, metadata.LogsStability))
+}
+
+// createDefaultConfig creates the default configuration for the exporter.
+func createDefaultConfig() component.Config {
+	return &Config{
+		TimeoutSettings: exporterhelper.NewDefaultTimeoutSettings(),
+		QueueSettings:   exporterhelper.NewDefaultQueueSettings(),
+		RetrySettings:   exporterhelper.NewDefaultRetrySettings(),
+	}
+}
+
+// createLogsExporter creates a new log exporter based on this config.
+func createLogsExporter(
+	ctx context.Context,
+	params exporter.CreateSettings,
+	cfg component.Config,
+) (exporter.Logs, error) {
+	chronicleCfg, ok := cfg.(*Config)
+	if !ok {
+		return nil, consumererror.NewPermanent(errors.New("invalid config type"))
+	}
```

Comment:
nit: No need to wrap this in consumererror, as this is not a "ConsumeXXX" function.